packages/alchemy/src/middleware/gas-fees.ts

```diff
@@ -0,0 +1,56 @@
+import type { AlchemyProvider } from "../provider";
+
+export enum GasFeeStrategy {
+  DEFAULT = "DEFAULT",
+  FIXED = "FIXED",
+  BASE_FEE_PERCENTAGE = "BASE_FEE_PERCENTAGE",
+  PRIORITY_FEE_PERCENTAGE = "PRIORITY_FEE_PERCENTAGE",
+}
+
+export interface GasFeeMode {
+  strategy: GasFeeStrategy;
+  value: bigint;
+}
+
+export const withAlchemyGasFeeEstimator = (
+  provider: AlchemyProvider,
+  feeMode: GasFeeMode
+): AlchemyProvider => {
+  if (feeMode.strategy === GasFeeStrategy.DEFAULT) {
+    return provider;
+  }
+
+  provider.withFeeDataGetter(async () => {
+    const block = await provider.rpcClient.getBlock({ blockTag: "latest" });
+    const baseFeePerGas = block.baseFeePerGas;
+    if (baseFeePerGas == null) {
+      throw new Error("baseFeePerGas is null");
+    }
+    const maxPriorityFeePerGas = BigInt(
+      await provider.rpcClient.getMaxPriorityFeePerGas()
+    );
+    // add 25% overhead to ensure mine
+    // because we're using bigint, 5n/4n = 1, so we need to scale by 100
```

Comment:
Really? Even if you multiply by 5 first and then divide by 4?

Reply:
pre-coffee brain fart... yea it works just fine with that order because math